tests/unit/test_avro_compatibility.py

```diff
@@ -36,11 +36,11 @@
     '{"type":"record","name":"myrecord","fields":[{"type":["null","string"],"name":"f1","default":'
     '"null"},{"type":"string","name":"f2","default":"foo"},{"type":"string","name":"f3","default":"bar"}]}'
 )
-invalidEnumDefaultValue = parse_avro_schema_definition(
-    '{"type": "enum", "name": "test_default", "symbols": ["A"], "default": "B"}'
+incorrectEnumDefaultValue = parse_avro_schema_definition(
+    '{"type": "enum", "name": "test_default", "symbols": ["A", "B"], "default": "B"}'
 )
 correctEnumDefaultValue = parse_avro_schema_definition(
-    '{"type": "enum", "name": "test_default", "symbols": ["A"], "default": "A"}'
+    '{"type": "enum", "name": "test_default", "symbols": ["A", "B"], "default": "A"}'
```

Comment:
As above, the "test objective" as desrcibed in in the test case is:
```
msg = "changing a default value in enum is a backward compatible change"
```
Considering that by "backward compatible" we mean "A new schema is backward compatible if it can be used to read the data written in a all previous schemas", I thought the new schema would need to accept "the same or more values" than previous schema. So if we had to define "B" as valid symbol for `incorrectEnumDefaultValue` I assumed this was also needed.

I just tested by not adding it, and the test passes, which I think exposes a bug: If old schema allow `A | B`, and new one only allows `A` ... any data containing `B` would be invalid

Reply:
The `A | B` to `A` sounds like an Avro defect.

Reply:
Hello!  I checked into this and the spec is _really_ confusing :/  I had to go back to the PR discussion to reason through what it's saying.

*TL;DR(1)*: Deleting an enum symbol is a permitted, but non-backwards compatible evolution.  If you never encounter `B`, everything's fine.

*TL;DR(2)*: Deleting an enum symbol with a default IS a backwards compatible evolution.  If you encounter `B`, the default is used.